File: fonte/U0 - Nivelamento/c/recursividade/chamandometodo.c
#include <stdio.h> 
#include <stdlib.h> 

void terceiro(){
   printf("3o - inicio e fim\n");
}

void segundo(){
   printf("2o - inicio\n");
   terceiro();
   printf("2o - fim\n");
}

void primeiro(){
   printf("1o - inicio\n");
   segundo();
   printf("1o - fim\n");
}

int main(int argc, char *argv[]) { 
   printf("main - inicio\n");
   primeiro();
   printf("main - fim\n");

   return 0; 
}

File: fonte/u00 Nivelamento/java/recursividade/ChamandoMetodo.java
class ChamandoMetodo {
   public static void primeiro(){
      System.out.println("1o - inicio");
      segundo();
      System.out.println("1o - fim");
   }
   public static void segundo(){
      System.out.println("2o - inicio");
      terceiro();
      System.out.println("2o - fim");
   }
   public static void terceiro(){
      System.out.println("3o - inicio e fim");
   }
   public static void main (String[] args){
      System.out.println("main - inicio");
      primeiro();
      System.out.println("main - fim");
   }
}
